client/java/src/main/java/org/jolokia/client/request/ValidatingResponseExtractor.java

```diff
@@ -54,10 +54,14 @@ public ValidatingResponseExtractor(int... pCodesAllowed) {
     public <RESP extends J4pResponse<REQ>, REQ extends J4pRequest> RESP extract(REQ pRequest,
                                                                       JSONObject pJsonResp)
             throws J4pRemoteException {
-
-        int status = pJsonResp.containsKey("status") ?
-                ((Long) pJsonResp.get("status")).intValue() :
-                0;
+	
+	int status = 0;
+	if( pJsonResp.containsKey("status")) {
+		Object o = pJsonResp.get("status");
+		if( o instanceof Long ) {
+			status = ((Long)o).intValue();
+		}
```

Comment:
As currently coded, it isn't going to be silently ignored. The int status will maintain it's 0 value, which will cause the J4pRemoteException below to be thrown, since 0 is not a valid response code in allowedCodes. 

So I think the current patch is correct, but if you wanted to be explicit and duplicate the throw, then I'm sure that's fine by me too.